lib/puma/launcher.rb

```diff
@@ -539,5 +530,25 @@ def log_config
 
       log "\n"
     end
+
+    def original_bundler_env
+      return unless defined?(Bundler)
+
+      env = Bundler::ORIGINAL_ENV.dup
+      # add -rbundler/setup so we load from Gemfile when restarting
+      bundle = "-rbundler/setup"
+      env["RUBYOPT"] = [env["RUBYOPT"], bundle].join(" ").lstrip unless env["RUBYOPT"].to_s.include?(bundle)
+      env
+    end
+
+    def update_env(previous_env)
+      replacement_env = if @options[:reload_env_on_restart]
+        ENV.to_h
```

Comment:
If I understand correctly, if this option is enabled, we'll fetch from the current `ENV` right?

Reply:
I believe so, unless there's some weirdness around the `ENV` constant itself that I'm not aware of.

Reply:
After trying to write an integration test for this, I'm confused on this would work.  It previously had `ENV.replace(previous_env)`.  I'm wondering if this feature would be accomplished by simply not calling `ENV.replace(previous_env)`?  It seems like `ENV` references the current environment variables.  Or would we need to somehow obtain the new environment variables?  Like loading them from another location?

Reply:
I was wondering how reloading ENV's during a hot restart would work in terms of how a user would update the `ENV`.  Could a user utilize `on_restart` to accomplish this and read new ENV's from a file?  Then maybe we don't need any code changes?
```
on_restart do
  ENV["TEST"] = "test2"
end
```

Reply:
Good point.

Maybe we would open a new process, have that process spit its `ENV` out so we can read it, then use `ENV.update` to read that all in?